src/dmd/statementsem.d

```diff
@@ -2683,22 +2683,32 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
                 ed = ds.isEnumDeclaration();
             if (ed)
             {
-              Lmembers:
+                bool missingMembers = false;
+            Lmembers:
                 foreach (es; *ed.members)
                 {
                     EnumMember em = es.isEnumMember();
                     if (em)
                     {
                         foreach (cs; *ss.cases)
                         {
-                            if (cs.exp.equals(em.value) || (!cs.exp.type.isString() && !em.value.type.isString() && cs.exp.toInteger() == em.value.toInteger()))
+                            if (cs.exp.equals(em.value) || (!cs.exp.type.isString() &&
+                                !em.value.type.isString() && cs.exp.toInteger() == em.value.toInteger()))
                                 continue Lmembers;
                         }
-                        ss.error("`enum` member `%s` not represented in `final switch`", em.toChars());
-                        sc.pop();
-                        return setError();
+                        if (!missingMembers)
+                        {
+                            missingMembers = true;
+                            ss.error("missing cases for `enum` members in `final switch`:");
+                        }
+                        ss.errorSupplemental("`%s`", em.toChars());
```

Comment:
`enum TOK` has 235 members.

If I were to do:
```
final switch (tok)
{
    case TOK.null_:
        break;
}
```
The resultant compiler error would not be desirable.

Reply:
Should I limit them to e.g. 10?

Reply:
I think the default is 6, or unlimited if `-v`(?) is passed. Take a look at the code that prints possible intended template matches.

Reply:
Ahh there it is https://github.com/dlang/dmd/blob/70482f4d647acf6978ce74fd804937f1008c523d/src/dmd/dtemplate.d#L5967 copy that logic

Reply:
I've done that now, but I don't know how to write a test for the verbose case since it puts absolute paths into the output, e.g.:
```
import    object	(/media/data1000/DebianHome/repos/dmd/test/../../druntime/import/object.d)
import    core.attribute	(/media/data1000/DebianHome/repos/dmd/test/../../druntime/import/core/attribute.d)
```

Reply:
`$p` should help, see the [README](https://github.com/dlang/dmd/blob/master/test/README.md):

> $p:<tail>$      paths ending with <tail> (which must refer to an existing file or directory)

Reply:
TRANSFORM_OUTPUT is better since I don't want the test to assert a specific order of semantic analysis. I'll steal the regex from fail15616b.d:
```
TRANSFORM_OUTPUT: remove_lines("^(predefs|binary|version|config|DFLAG|parse|import|semantic|entry|\s*$)")
```